src/browser-pool.js

```diff
@@ -202,16 +205,17 @@ class BrowserPool extends EventEmitter {
      * @return {Promise<BrowserController>}
      * @private
      */
-    async _launchBrowser() {
+    async _launchBrowser(options) {
         const browserPlugin = this._pickNewBrowserPluginToLaunch();
-        const browserControllerContext = await browserPlugin.createBrowserControllerContext();
+        const launchContext = browserPlugin.createLaunchContext(options);
+        launchContext.browserPlugin = browserPlugin;
 
-        await this._executeHooks(this.preLaunchHooks, browserPlugin, browserControllerContext);
+        await this._executeHooks(this.preLaunchHooks, launchContext);
```

Comment:
I think we should keep the `plugin` there. Maybe swap the params (`launchContext`, `browserPlugin`). Launch context does not have a reference to the plugin, no? I also think we should add a reference to the plugin to the `browserController`.

Reply:
I have added the plugin instance to the launchContext. More precisely move it from the browser controller top level to the launch context.